src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -2,58 +2,241 @@
 // Licensed under the MIT License.
 using System.Threading;
 using System.Threading.Tasks;
+using Bicep.Core.FileSystem;
 using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Completions;
 using Bicep.LanguageServer.Extensions;
 using Bicep.LanguageServer.Providers;
 using Bicep.LanguageServer.Utils;
 using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
 using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using System;
+using Bicep.Core.Parsing;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using System.Linq;
 
 namespace Bicep.LanguageServer.Handlers
 {
     public class BicepDefinitionHandler : DefinitionHandlerBase
     {
         private readonly ISymbolResolver symbolResolver;
 
-        public BicepDefinitionHandler(ISymbolResolver symbolResolver) : base()
+        private readonly ICompilationManager compilationManager;
+
+        private readonly IFileResolver fileResolver;
+
+        public BicepDefinitionHandler(ISymbolResolver symbolResolver, ICompilationManager compilationManager, IFileResolver fileResolver) : base()
         {
             this.symbolResolver = symbolResolver;
+            this.compilationManager = compilationManager;
+            this.fileResolver = fileResolver;
         }
 
         public override Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken)
         {
+            var context = this.compilationManager.GetCompilation(request.TextDocument.Uri);
+            if (context == null)
```

Comment:
>== [](http://example.com/codeflow?start=24&length=2)

nit - anywhere we have `== null`, we should use `is null` unless we really want `Equals()` to be called.

Reply:
Right just read up on a primer on this, good to keep in mind